### PR TITLE
net6.0 support

### LINKED ISSUE #165: Add support for .NET 6
Given that the final stable release of .NET 6 is just around the corner (and indeed may have already happened by the time this issue gets the appropriate attention), these rules need to add support for the .NET 6 target framework.

### PR DESCRIPTION
**Pull request by omsmith**
Closes: https://github.com/Brightspace/rules_csharp/issues/165

### CODE REVIEW

File: csharp/private/frameworks/net60.BUILD [hunk 1]
```diff
@@ -0,0 +1,1442 @@
+# This file was generated by build\src\D2L.Build.BazelGenerator\Commands\GenerateFrameworkBuildFile.cs
+# Don't edit it by hand! You should regenerate this if the .NET package gets updated.
+
+load("@d2l_rules_csharp//csharp:defs.bzl", "csharp_library_set", "import_library")
+
+csharp_library_set(
+    name = "StandardReferences",
```
Comment (omsmith) [inline csharp/private/frameworks/net60.BUILD:7]: I forget if this list is just always everything? Generated the list from other target names.

File: csharp/private/frameworks/net60.BUILD [hunk 2]
```diff
@@ -0,0 +1,1442 @@
...
+        "//:WindowsBase",
+    ],
+    visibility = ["@net//:__pkg__"],
+)
+
+import_library(
+    name = "Microsoft.CSharp",
+    target_framework = "net6.0",
+    refdll = ":ref/net6.0/Microsoft.CSharp.dll",
+    deps = [":mscorlib"],
```
Comment (omsmith) [inline csharp/private/frameworks/net60.BUILD:177]: Generator spit out mscorlib for everything. Don't recall if we manually changed that for previous frameworks.

File: csharp/private/net/BUILD [hunk 3]
```diff
@@ -389,6 +393,7 @@ import_multiframework_library(
     net472 = "@net472//:Microsoft.Win32.Primitives",
     net48 = "@net48//:Microsoft.Win32.Primitives",
     net5_0 = "@net5.0//:Microsoft.Win32.Primitives",
+    net6_0 = "@net6.0//:Microsoft.Win32.Primitives",
```
Comment (omsmith) [inline csharp/private/net/BUILD:396]: Automatically added net6_0 everywhere we had net5_0

File: csharp/private/sdk.bzl [hunk 4]
```diff
@@ -3,21 +3,21 @@ Declarations for the .NET SDK Downloads URLs and version
...
+        "url": "https://download.visualstudio.microsoft.com/download/pr/20283373-1d83-4879-8278-0afb7fd4035e/56f204f174743b29a656499ad0fc93c3/dotnet-sdk-6.0.100-rc.2.21505.57-linux-x64.tar.gz",
+        "hash": "506752857752512f199e97827c1e102656dc6490585758a93f37a792de2f9461",
     },
     "osx": {
-        "url": "https://download.visualstudio.microsoft.com/download/pr/1dfb70ed-2f4a-40fb-a920-4a866440896c/0c4a206d58dc75417217883e8271a633/dotnet-sdk-5.0.104-osx-x64.tar.gz",
-        "hash": "cf2a927fc61ef22e72a74a021c5352db21680550292ab879f93e63ce4a895fe4",
+        "url": "https://download.visualstudio.microsoft.com/download/pr/35655ed6-3e37-4fa3-8990-5c1827469ce5/f9f920ff05b0aa5961a8b30e2824de7d/dotnet-sdk-6.0.100-rc.2.21505.57-osx-x64.tar.gz",
+        "hash": "b42a7b378166bee735523f3ee656adbcc28ec6c0bc103ab1379e81a2fbd191f5",
     },
 }
```
Comment (omsmith) [inline csharp/private/sdk.bzl:20]: RC, update PR once GA

**Comment by omsmith** [comment]
Its been a while, but I think I got it?

FYI @j3parker @jimevans

**Review by j3parker — DISMISSED** [review]

File: csharp/private/frameworks/net60.BUILD (hunk 1 above)
Comment (omsmith) [reply csharp/private/frameworks/net60.BUILD:7]: Compared to the list passed to csc for an empty project. Same items, different ordering.

```diff
diff --git a/csharp/private/frameworks/net60.BUILD b/csharp/private/frameworks/net60.BUILD
index 76019a6..171b0ee 100644
--- a/csharp/private/frameworks/net60.BUILD
+++ b/csharp/private/frameworks/net60.BUILD
@@ -8,32 +8,31 @@ csharp_library_set(
     target_framework = "net6.0",
     deps = [
         "//:Microsoft.CSharp",
-        "//:Microsoft.VisualBasic",
         "//:Microsoft.VisualBasic.Core",
+        "//:Microsoft.VisualBasic",
         "//:Microsoft.Win32.Primitives",
         "//:Microsoft.Win32.Registry",
         "//:mscorlib",
         "//:netstandard",
-        "//:System",
         "//:System.AppContext",
         "//:System.Buffers",
-        "//:System.Collections",
         "//:System.Collections.Concurrent",
+        "//:System.Collections",
         "//:System.Collections.Immutable",
         "//:System.Collections.NonGeneric",
         "//:System.Collections.Specialized",
-        "//:System.ComponentModel",
         "//:System.ComponentModel.Annotations",
         "//:System.ComponentModel.DataAnnotations",
+        "//:System.ComponentModel",
         "//:System.ComponentModel.EventBasedAsync",
         "//:System.ComponentModel.Primitives",
         "//:System.ComponentModel.TypeConverter",
         "//:System.Configuration",
         "//:System.Console",
         "//:System.Core",
-        "//:System.Data",
         "//:System.Data.Common",
         "//:System.Data.DataSetExtensions",
+        "//:System.Data",
         "//:System.Diagnostics.Contracts",
         "//:System.Diagnostics.Debug",
         "//:System.Diagnostics.DiagnosticSource",
@@ -44,27 +43,28 @@ csharp_library_set(
         "//:System.Diagnostics.Tools",
         "//:System.Diagnostics.TraceSource",
         "//:System.Diagnostics.Tracing",
+        "//:System",
         "//:System.Drawing",
         "//:System.Drawing.Primitives",
         "//:System.Dynamic.Runtime",
         "//:System.Formats.Asn1",
-        "//:System.Globalization",
         "//:System.Globalization.Calendars",
+        "//:System.Globalization",
         "//:System.Globalization.Extensions",
-        "//:System.IO",
-        "//:System.IO.Compression",
         "//:System.IO.Compression.Brotli",
+        "//:System.IO.Compression",
         "//:System.IO.Compression.FileSystem",
         "//:System.IO.Compression.ZipFile",
-        "//:System.IO.FileSystem",
+        "//:System.IO",
         "//:System.IO.FileSystem.AccessControl",
+        "//:System.IO.FileSystem",
         "//:System.IO.FileSystem.DriveInfo",
         "//:System.IO.FileSystem.Primitives",
         "//:System.IO.FileSystem.Watcher",
         "//:System.IO.IsolatedStorage",
         "//:System.IO.MemoryMappedFiles",
-        "//:System.IO.Pipes",
         "//:System.IO.Pipes.AccessControl",
+        "//:System.IO.Pipes",
         "//:System.IO.UnmanagedMemoryStream",
         "//:System.Linq",
         "//:System.Linq.Expressions",
@@ -87,13 +87,13 @@ csharp_library_set(
         "//:System.Net.WebClient",
         "//:System.Net.WebHeaderCollection",
         "//:System.Net.WebProxy",
-        "//:System.Net.WebSockets",
         "//:System.Net.WebSockets.Client",
+        "//:System.Net.WebSockets",
         "//:System.Numerics",
         "//:System.Numerics.Vectors",
         "//:System.ObjectModel",
-        "//:System.Reflection",
         "//:System.Reflection.DispatchProxy",
+        "//:System.Reflection",
         "//:System.Reflection.Emit",
         "//:System.Reflection.Emit.ILGeneration",
         "//:System.Reflection.Emit.Lightweight",
@@ -104,9 +104,9 @@ csharp_library_set(
         "//:System.Resources.Reader",
         "//:System.Resources.ResourceManager",
         "//:System.Resources.Writer",
-        "//:System.Runtime",
         "//:System.Runtime.CompilerServices.Unsafe",
         "//:System.Runtime.CompilerServices.VisualC",
+        "//:System.Runtime",
         "//:System.Runtime.Extensions",
         "//:System.Runtime.Handles",
         "//:System.Runtime.InteropServices",
@@ -119,7 +119,6 @@ csharp_library_set(
         "//:System.Runtime.Serialization.Json",
         "//:System.Runtime.Serialization.Primitives",
         "//:System.Runtime.Serialization.Xml",
-        "//:System.Security",
         "//:System.Security.AccessControl",
         "//:System.Security.Claims",
         "//:System.Security.Cryptography.Algorithms",
@@ -129,22 +128,23 @@ csharp_library_set(
         "//:System.Security.Cryptography.OpenSsl",
         "//:System.Security.Cryptography.Primitives",
         "//:System.Security.Cryptography.X509Certificates",
+        "//:System.Security",
         "//:System.Security.Principal",
         "//:System.Security.Principal.Windows",
         "//:System.Security.SecureString",
         "//:System.ServiceModel.Web",
         "//:System.ServiceProcess",
-        "//:System.Text.Encoding",
         "//:System.Text.Encoding.CodePages",
+        "//:System.Text.Encoding",
         "//:System.Text.Encoding.Extensions",
         "//:System.Text.Encodings.Web",
         "//:System.Text.Json",
         "//:System.Text.RegularExpressions",
-        "//:System.Threading",
         "//:System.Threading.Channels",
+        "//:System.Threading",
         "//:System.Threading.Overlapped",
-        "//:System.Threading.Tasks",
         "//:System.Threading.Tasks.Dataflow",
+        "//:System.Threading.Tasks",
         "//:System.Threading.Tasks.Extensions",
         "//:System.Threading.Tasks.Parallel",
         "//:System.Threading.Thread",
```

File: csharp/private/sdk.bzl [hunk 5]
```diff
@@ -3,21 +3,21 @@ Declarations for the .NET SDK Downloads URLs and version
...
 
-RUNTIME_TFM = "netcoreapp5.0"
-RUNTIME_FRAMEWORK_VERSION = "5.0.0"
+RUNTIME_TFM = "net6.0"
+RUNTIME_FRAMEWORK_VERSION = "6.0.0"
```
[the remark this answers is not in the document]
Comment (omsmith) [reply csharp/private/sdk.bzl:23]: Expect it to be correct once we're GA.

```json
{
  "runtimeOptions": {
    "tfm": "net6.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "6.0.0-rc.2.21480.5"
    },
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
    }
  }
}
```

[the remark this answers is not in the document]
Comment (omsmith) [reply csharp/private/sdk.bzl:23]: Validated with VS2022 17.1.0 Preview

```json
{
  "runtimeOptions": {
    "tfm": "net6.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "6.0.0"
    },
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
    }
  }
}
```

**Review by j3parker — APPROVED** [review]

**Pull request merged**